pkg/interceptors/server/server_test.go

```diff
@@ -274,3 +274,15 @@ func Test_UpdateCRDWithCaCert(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func Test_CheckCertValidity(t *testing.T) {
+	// Adding delay of a minute so that ticker will be triggered
+	time.Sleep(time.Minute)
```

Comment:
I think if we could pass in the ticker via a parameter, we could better test just the checkCertValidity function

Reply:
Another option would be to pass in a [clock.Ticker](https://pkg.go.dev/k8s.io/utils@v0.0.0-20220728103510-ee6ede2d64ed/clock#Ticker) to the function above and then use https://pkg.go.dev/k8s.io/utils/clock/testing#FakeClock.NewTicker in the test